mypy/stubtest.py

```diff
@@ -310,15 +341,22 @@ def verify_mypyfile(
         if not o.module_hidden and (not is_probably_private(m) or hasattr(runtime, m))
     }
 
+    imported_symbols = _get_imported_symbol_names(runtime)
+
     def _belongs_to_runtime(r: types.ModuleType, attr: str) -> bool:
         obj = getattr(r, attr)
-        try:
-            obj_mod = getattr(obj, "__module__", None)
-        except Exception:
+        if isinstance(obj, types.ModuleType):
             return False
-        if obj_mod is not None:
-            return bool(obj_mod == r.__name__)
-        return not isinstance(obj, types.ModuleType)
+        if callable(obj):
+            try:
+                obj_mod = getattr(obj, "__module__", None)
+            except Exception:
+                return False
+            if obj_mod is not None:
+                return bool(obj_mod == r.__name__)
+        if imported_symbols is not None:
+            return attr not in imported_symbols
```

Comment:
I guess the other change in behaviour here is that if it's not callable and we don't have a symtable, we no longer look at the `__module__` attribute. This seems fine to me, based on the things that`__module__` is documented as existing on.